src/lib/Microsoft.Health.Extensions.Fhir.R4/Service/ResourceManagementService.cs

```diff
@@ -63,6 +67,14 @@ protected static async Task<TResource> CreateResourceByIdentityAsync<TResource>(
 
             propertySetter?.Invoke(resource, identifier);
 
+            // If patient resource, then generate id programatically and issue an update to FHIR service
+            // The patient resource will be created if the resource with the given id does not already exist.
+            if (typeof(TResource) == typeof(Model.Patient))
+            {
+                resource.Id = ToGuid(identifier.Value.ToString()).ToString();
+                return await client.UpdateAsync<TResource>(resource).ConfigureAwait(false);

```

Comment:
Have we verified the behavior of the update will cause the collision on the second "create"?

Reply:
In theory one request should 'win' and create the patient with the deterministic id and whichever request comes second should update the resource. I partially verified this behavior by modifying the search always return null so that we always create a patient on each request, and verified that on the 2nd request we just update the existing patient resource that was created in the 1st request.

I have not been able to come up with a scenario where the 2nd request fails, but if it did I assume the code would retry and we'd just update the existing patient resource? I do not think there is a scenario where we end up creating two patient resources.